src/parse/rules/java_rules.build_defs

```diff
@@ -2,6 +2,7 @@
 
 
 def java_library(name:str, srcs:list=None, src_dir:str=None, resources:list=None, resources_root:str=None,
+                 source_level:str=None, target_level:str=None, release_level:str=None,
```

Comment:
You can default these - e.g. `source_level:str=CONFIG.JAVA_SOURCE_LEVEL` which would simplify your ifs below